### PR TITLE
Jd/fix cost functions

### PR DESCRIPTION
**Pull request by jd-lara**
This PR adds new cases for cost function implementation testing

### CODE REVIEW

File: src/library/psitest_library.jl [hunk 1]
```diff
@@ -4315,3 +4129,154 @@ function build_c_sys5_radial(; raw_data, kwargs...)
...
+    PSY.add_component!(cost_test_sys, load)
+    PSY.add_component!(cost_test_sys, gen)
+    PSY.add_time_series!(cost_test_sys, load, load_forecast)
+    return cost_test_sys
+end
+
+function build_linear_cost_test_sys(; kwargs...)
+    base_sys = _build_cost_base_test_sys(; kwargs...)
+    node = PSY.get_component(ACBus, base_sys, "nodeA")
+    test_gen = thermal_generator_linear_cost(node)
```
Comment (GabrielKS) [inline src/library/psitest_library.jl:4185]: I don't see where this `thermal_generator_linear_cost` is defined, same for the similar ones below

File: src/library/psitest_library.jl [hunk 2]
```diff
@@ -4315,3 +4129,154 @@ function build_c_sys5_radial(; raw_data, kwargs...)
...
+    test_gen = thermal_generator_quad_cost(node)
+    PSY.add_component!(base_sys, test_gen)
+    return base_sys
+end
+
+function build_quadratic_fuel_test_sys(; kwargs...)
+    base_sys = _build_cost_base_test_sys(; kwargs...)
+    node = PSY.get_component(ACBus, base_sys, "nodeA")
+    test_gen = thermal_generator_quad_fuel(node)
+
```
Comment (GabrielKS) [inline src/library/psitest_library.jl:4210]: missing `add_component!`?

File: src/library/psitest_library.jl [hunk 3]
```diff
@@ -4315,3 +4129,154 @@ function build_c_sys5_radial(; raw_data, kwargs...)
...
+function build_linear_cost_test_sys(; kwargs...)
+    base_sys = _build_cost_base_test_sys(; kwargs...)
+    node = PSY.get_component(ACBus, base_sys, "nodeA")
+    test_gen = thermal_generator_linear_cost(node)
+    PSY.add_component!(base_sys, test_gen)
+    return base_sys
+end
+
+function build_linear_fuel_test_sys(; kwargs...)
```
Comment (GabrielKS) [inline src/library/psitest_library.jl:4190]: Nitpick: could eliminate duplication by defining a helper function
```julia
function build_cost_test_sys(gen_func; kwargs...)
    ...
    test_gen = gen_func(node)
    ...
end
```
(or fold this into `_build_cost_base_test_sys`) and then each of these is like
```suggestion
build_linear_fuel_test_sys(; kwargs...) =
    build_cost_test_sys(thermal_generator_linear_fuel; kwargs...)
```

File: src/system_library.jl [hunk 4]
```diff
@@ -1,3 +1,4 @@
+include(joinpath(DATA_DIR, "psy_data", "generation_cost_function_data.jl"))
```
Comment (GabrielKS) [inline src/system_library.jl:1]: Missing this file

**Review by GabrielKS — CHANGES_REQUESTED** [review]
~~I think you're missing some code~~ looks good aside from some duplication and maybe a bug thereby caused

File: src/library/psitest_library.jl (hunk 1 above)
Comment (jd-lara) [reply src/library/psitest_library.jl:4185]: These are defined in TestData

File: src/system_library.jl (hunk 4 above)
Comment (jd-lara) [reply src/system_library.jl:1]: This is in the TestData repo

**Comment by GabrielKS** [comment]
Depends on https://github.com/NREL-Sienna/PowerSystemsTestData/pull/54

File: src/library/psitest_library.jl (hunk 2 above)
Comment (jd-lara) [reply src/library/psitest_library.jl]: fixed

**Review by rodrigomha — APPROVED** [review]
This looks good. I can take care of testing all of these in https://github.com/NREL-Sienna/PowerSimulations.jl/pull/1104

**Review by GabrielKS — APPROVED** [review]

**Comment by jd-lara** [comment]
I'll merge, and if we need more, we should open more. I don't want to make these very large PRs a common thing. 

@rodrigomha, note that in some cases, there is non-sensical existing data already

**Pull request merged**